pilot/pkg/model/context.go

```diff
@@ -220,10 +225,13 @@ func ParseMetadata(metadata *types.Struct) map[string]string {
 	return res
 }
 
-// ParseServiceNode is the inverse of service node function
-func ParseServiceNode(s string) (*Proxy, error) {
+// ParseServiceNodeWithMetadata parse the Envoy Node from the string generated by ServiceNode
+// fuction and the metadata.
+func ParseServiceNodeWithMetadata(s string, metadata map[string]string) (*Proxy, error) {
```

Comment:
Would not change this name, but pass in a `*core.Node` param.

Maybe a following up .

Reply:
OK, I'll leave it as it right now.